test/remote/gateways/remote_pay_trace_test.rb

```diff
@@ -0,0 +1,201 @@
+    assert_success auth
+
+    assert capture = @gateway.capture(auth.authorization, @options.merge(amount: 3.00))
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture('')
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_refund
+    purchase = @gateway.purchase(200, @credit_card, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization, @options.merge(amount: 1.00))
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_partial_refund
+    purchase = @gateway.purchase(400, @mastercard, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization, @options.merge(amount: 3.00))
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_refund_without_amount
+    purchase = @gateway.purchase(@amount, @discover, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization)
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_failed_refund
+    response = @gateway.refund('', @options)
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_void
+    auth = @gateway.authorize(@amount, @amex, @options)
+    assert_success auth
+
+    assert void = @gateway.void(auth.authorization)
+    assert_success void
+    assert_equal 'Your transaction was successfully voided.', void.message
+  end
+
+  def test_failed_void
+    response = @gateway.void('')
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@mastercard, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_store_and_redact_customer_profile
+    response = @gateway.store(@mastercard, @options)
+    assert_success response
+    customer_id = response.params['customer_id']
+    redact = @gateway.redact(customer_id)
+    assert_success redact
+    assert_equal true, redact.success?
+  end
+
+  def test_duplicate_customer_creation
+    create = @gateway.store(@discover, @options)
+    customer_id = create.params['customer_id']
+    response = @gateway.store(@discover, @options.merge(customer_id: customer_id))
+    assert_failure response
+    assert_equal false, response.success?
```

Comment:
I think this may be superfluous in conjunction with `assert_failure response`; see https://github.com/activemerchant/active_merchant/blob/a9c588f90c4c1d59f7b0f5e1dc40b9ca12cb7fbf/test/test_helper.rb#L78

Reply:
I think that you are definitely correct. Will remove!